### PR TITLE
DEP: Deprecate with replacement CCITParameters

### PR DESCRIPTION
**Pull request by j-t-1**
Deprecate CCITParameters with replacement CCITTParameters.

### CODE REVIEW

**Comment by stefan6419846** [comment]
Please make sure to keep coverage by ensuring that the old name still works as intended.

File: pypdf/filters.py [hunk 1]
```diff
@@ -501,6 +502,19 @@ def group(self) -> int:
         return CCITTgroup
 
 
+def __create_old_class_instance(
+    K: int = 0,
+    columns: int = 0,
+    rows: int = 0
+) -> CCITTParameters:
+    deprecate_with_replacement("CCITParameters", "CCITTParameters", "5.0.0")
```
Comment (stefan6419846) [inline pypdf/filters.py:510]: The message looks wrong:

> CCITParameters is deprecated and will be removed in pypdf 5.0.0. Use CCITTParameters instead.

The current release is 5.1.0.

Comment (j-t-1) [reply pypdf/filters.py:510]: Is obvious (now!). Will change to 5.2.0. Plus add a test of the original.

**Comment by j-t-1** [comment]
> Please make sure to keep coverage by ensuring that the old name still works as intended.

Added. The check on 511 is indirect, so I think is okay now.

File: pypdf/filters.py (hunk 1 above)
Comment (stefan6419846) [reply pypdf/filters.py]: Shouldn't it still be 6.0.0 instead? We will surely not remove this in the next patch release.

**Review by stefan6419846 — APPROVED** [review]

**Pull request merged**